### PR TITLE
Add pre-release test suite

### PR DESCRIPTION
**Pull request by Katka92**
### What does this PR do?
Adding variable ACCOUNT_ENV to docker entrypoint. Adding a pre-release test suite.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/16549

### How have you tested this PR?
Localy.

### CODE REVIEW

File: dockerfiles/e2e-saas/docker-entrypoint.sh [hunk 1]
```diff
@@ -59,13 +59,23 @@ length=${#USERNAME}
...
+  else
+    preview=""
+  fi
+else 
+  if [ "$ACCOUNT_ENV" == "prod" ]; then
+    preview=""
+  else 
+    preview="prod-preview."
+  fi
 fi
```
Comment (ibuziuk) [inline dockerfiles/e2e-saas/docker-entrypoint.sh:79]: this logic is not straight forward for me. I believe we can do a better job of determining staging vs prod environment + would be nice to extract it to a separate function

Comment (Katka92) [reply dockerfiles/e2e-saas/docker-entrypoint.sh:79]: I can definitely separate that to a function. Do you have any idea how to improve this logic? I'm not aware of any other check.

Comment (ibuziuk) [reply dockerfiles/e2e-saas/docker-entrypoint.sh:79]: first `ACCOUNT_ENV variable is not set, retrieving env from URL` is it really true? according to the logic preview vs prod is retrieved based on the username?

File: dockerfiles/e2e-saas/docker-entrypoint.sh [hunk 2]
```diff
@@ -59,13 +59,23 @@ length=${#USERNAME}
...
 
 response=$(curl -s -g -X GET --header 'Accept: application/json' "https://api.${preview}openshift.io/api/users?filter[username]=$USERNAME")
```
Comment (ibuziuk) [inline dockerfiles/e2e-saas/docker-entrypoint.sh:81]: in general, I'm not a big fan of the manipulations with `${preview}`

Comment (ibuziuk) [reply dockerfiles/e2e-saas/docker-entrypoint.sh:81]: I believe we can simply determine URLs based on the `ACCOUNT_ENV` and avoid using extra preview env var

File: dockerfiles/e2e-saas/docker-entrypoint.sh (hunk 1 above)
Comment (Katka92) [reply dockerfiles/e2e-saas/docker-entrypoint.sh:79]: Yeah, it is retrieved from username. I'll fix that. Theoretically, we can remove that logic and just go with the mandatory variable `ACCOUNT_ENV` that would specify the env. But that would require changes in more scripts. If you want, I can do that as a part of this PR.

**Pull request merged**